packages/react/src/components/RuleBuilder/RuleBuilderEditor.jsx

```diff
@@ -0,0 +1,146 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+
+import { TableColumnsPropTypes } from '../Table/TablePropTypes';
+
+import Rule, { RuleGroupPropType } from './Rule';
+import RuleBuilderHeader from './RuleBuilderHeader';
+import {
+  addRule,
+  filterRulesById,
+  findRulePathById,
+  generateRuleGroup,
+  getRuleByPath,
+  updateRuleAtPath,
+} from './utils';
+
+const propTypes = {
+  // the rules passed into the component. The RuleBuilder is a controlled component, so
+  // this works the same as passing defaultValue to a controlled input component.
+  defaultRules: RuleGroupPropType,
+  // the columns of the table to be used in the column dropdown.
+  columns: TableColumnsPropTypes.isRequired,
+  onChange: PropTypes.func,
```

Comment:
what does this event look like

Reply:
added an onChange action to the stories, so show the rule tree.

Reply:
you can document the onChange here, since, this is what people will see
```
/* Function accepting the current rule */
onChange: PropTypes.func
```
ie, don't be afraid to add comments to the code... they are useful ;)  The comment for coumns is fine, but event handled should be documented in the code as to what they accept.

Reply:
Got it. It's honestly been a few years since using prop-types, so I'm a bit out of the game. I've been used to being able to define callbacks and their params in TypeScript. I'll go through and add some comments to more thoroughly document these events.

Reply:
addressed in 2899f10 push.